mewarpx/examples/diagnostic_plotting.py

```diff
@@ -0,0 +1,196 @@
+# DIAG_INTERVAL = 100.0 * DT # 32 / FREQ
+
+# --- Number of time steps
+max_steps = int(TOTAL_TIME / DT)
+diag_steps = 2
+diagnostic_intervals = "::1"
+
+print('Setting up simulation with')
+print('  dt = %.3e s' % DT)
+print('  Total time = %.3e s (%i timesteps)' % (TOTAL_TIME, max_steps))
+
+##########################
+# physics components
+##########################
+
+anode_voltage = lambda t: VOLTAGE * np.sin(2.0 * np.pi * FREQ * t)
+
+v_rms_elec = np.sqrt(constants.kb * T_ELEC / constants.m_e)
+v_rms_ion = np.sqrt(constants.kb * T_INERT / M_ION)
+
+uniform_plasma_elec = picmi.UniformDistribution(
+    density = PLASMA_DENSITY,
+    upper_bound = [None] * 3,
+    rms_velocity = [v_rms_elec] * 3,
+    directed_velocity = [0.] * 3
+)
+
+uniform_plasma_ion = picmi.UniformDistribution(
+    density = PLASMA_DENSITY,
+    upper_bound = [None] * 3,
+    rms_velocity = [v_rms_ion] * 3,
+    directed_velocity = [0.] * 3
+)
+
+electrons = picmi.Species(
+    particle_type='electron', name='electrons',
+    initial_distribution=uniform_plasma_elec
+)
+ions = picmi.Species(
+    particle_type='He', name='he_ions',
+    charge='q_e',
+    initial_distribution=uniform_plasma_ion
+)
+
+# MCC collisions
+mcc_wrapper = MCC(
+    electrons, ions, T_INERT=T_INERT, N_INERT=N_INERT,
+    exclude_collisions=['charge_exchange']
+)
+
+##########################
+# numerics components
+##########################
+
+grid = picmi.Cartesian2DGrid(
+    number_of_cells = [nx, nz],
+    lower_bound = [xmin, zmin],
+    upper_bound = [xmax, zmax],
+    lower_boundary_conditions=['periodic', 'dirichlet'],
+    upper_boundary_conditions=['periodic', 'dirichlet'],
+    lower_boundary_conditions_particles=['periodic', 'absorbing'],
+    upper_boundary_conditions_particles=['periodic', 'absorbing'],
+    warpx_potential_hi_z = "%.1f*sin(2*pi*%.5e*t)" % (VOLTAGE, FREQ),
+    moving_window_velocity = None,
+    warpx_max_grid_size = nz//4
+)
+
+##########################
+# declare solver
+##########################
+
+solver = picmi.ElectrostaticSolver(
+   grid=grid, method='Multigrid', required_precision=1
+)
+#solver = PoissonSolverPseudo1D(grid=grid)
+
+##########################
+# diagnostics
+##########################
+
+field_diag = picmi.FieldDiagnostic(
+    name = 'diags',
+    grid = grid,
+    period = diagnostic_intervals,
+    data_list = ['rho_electrons', 'rho_he_ions', 'phi'],
+    write_dir = 'diags/',
+)
+
+##########################
+# simulation setup
+##########################
+
+mwxrun.simulation.solver = solver
+mwxrun.simulation.time_step_size = DT
+mwxrun.simulation.max_steps = max_steps
+
+mwxrun.simulation.add_species(
+    electrons,
+    layout = picmi.GriddedLayout(
+        n_macroparticle_per_cell=number_per_cell_each_dim, grid=grid
+    )
+)
+mwxrun.simulation.add_species(
+    ions,
+    layout = picmi.GriddedLayout(
+        n_macroparticle_per_cell=number_per_cell_each_dim, grid=grid
+    )
+)
```

Comment:
Use the new mepicmi Species, which adds species to the simulation by default, without having to specify add_species separately.

Reply:
I think you pass `n_macroparticle_per_cell` and `grid` in the initial `mepicmi.Species` calls and then you can delete these `add_species` blocks entirely?